timeseries.go

```diff
@@ -5,21 +5,36 @@ import (
 )
 
 // TimeSeries represents an array of candles
-type TimeSeries struct {
-	Candles []*Candle
+type TimeSeries interface {
```

Comment:
I like the idea behind this, but, because the original `TimeSeries` struct was exported, I don't think we can just replace the struct def with an interface def, because it will break every project where someone was previously using a concrete `TimeSeries` struct. I think what we need to do is invert the naming here, s.t. the we give the interface another name (maybe `ITimeSeries` to borrow from C++, but I'm sure you can suggest something better) and keep `TimeSeries` bound to the struct def. WDYT?

Reply:
Yes, there must be a more elegant way. Sorry for bothering you too early. I need to think a bit more about it all.